src/coreclr/vm/jitinterface.cpp

```diff
@@ -7684,29 +7684,12 @@ static void getMethodInfoHelper(
                              (ftn->RequiresInstMethodTableArg() ? CORINFO_GENERICS_CTXT_FROM_METHODTABLE : 0) |
                              (ftn->RequiresInstMethodDescArg() ? CORINFO_GENERICS_CTXT_FROM_METHODDESC : 0)));
 
-    // EEJitManager::ResolveEHClause and CrawlFrame::GetExactGenericInstantiations
-    // need to be able to get to CORINFO_GENERICS_CTXT_MASK if there are any
-    // catch clauses like "try {} catch(MyException<T> e) {}".
-    // Such constructs are rare, and having to extend the lifetime of variable
-    // for such cases is reasonable
+    // Always keep generics content in case the exact generic instantiation needs to
```

Comment:
This will introduce performance regressions.

Reply:
I think the performance regressions will only happen when walking stack (EH or `new StackFrame()`) because it needs to parse more things from GCInfo? 

So there're some tradeoffs here. I can come up with the below options:

1. Use the exact generic instantiation: may introduce performance regressions (hopefully only happen when walking stack)
2. Use the canonical type: no performance regression, but it will leak the implementation details
3. Introduce a runtime knob: no performance regression and can be enabled only when necessary
4. Keep the original behavior: this would makes locating and identifying an exception from the stack trace hard

Reply:
> I think the performance regressions will only happen when walking stack

No. It will regress code quality. The JIT can optimize out the hidden parameter in many cases. Setting this flag unconditionally will force the JIT to always burn register or stack slot for the hidden parameter.

Reply:
We do not aim to provide perfect stacktraces for diagnostics. The stacktraces can have missing frames due to optimizations.

We could follow the same principle here: Print the extra info if it happened to survive optimizations, but do not de-optimize the code to make the extra info survive.

Reply:
How about only print generic arguments in cases `try {} catch(MyException<T> e) {}` where the JIT would set the flag?

Reply:
> The stacktraces can have missing frames due to optimizations.

Would it make sense to force this only for debug then?

Reply:
> Would it make sense to force this only for debug then?

I would expect the generic context to be preserved for debug code on the JIT side already.

Reply:
> How about only printing exact generic arguments when the JIT set the flag?

Right, it is what I meant - print the additional information only if it survived optimizations.